components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java

```diff
@@ -460,9 +460,9 @@ private MappingResource getMappingResource(MessageContext synCtx, String configK
      * @return Map filed with property name and the value
      */
     private Map getPropertiesMap(List<String> propertiesNamesList, MessageContext synCtx) {
-        Map<String, Map<String, String>> propertiesMap = new HashMap<>();
+        Map<String, Map<String, Object>> propertiesMap = new HashMap<>();
         String[] contextAndName;
-        String value = EMPTY_STRING;
+        Object value = EMPTY_STRING;
```

Comment:
In default we are using an empty string, thats why initialized to that. anyway even if we not initialize it according to the following logic it will be set to an empty string